keepassxc-browser/content/fields.js

```diff
@@ -391,18 +391,120 @@ kpxcFields.isVisible = function(elem) {
     if (elemStyle.visibility && (elemStyle.visibility === 'hidden' || elemStyle.visibility === 'collapse')
         || (opacity < MIN_OPACITY || opacity > MAX_OPACITY)
         || parseInt(elemStyle.width, 10) <= MIN_INPUT_FIELD_WIDTH_PX
-        || parseInt(elemStyle.height, 10) <= MIN_INPUT_FIELD_WIDTH_PX) {
+        || parseInt(elemStyle.height, 10) <= MIN_INPUT_FIELD_WIDTH_PX
+        || kpxcFields.isElementClipped(elemStyle) 
+    ) {
         return false;
     }
 
-    // Check for parent opacity
-    if (kpxcFields.traverseParents(elem, f => f.style.opacity === '0')) {
+    // Check for parent visibility
+    if (kpxcFields.traverseParents(elem, (f) => {
+        const fStyle = getComputedStyle(f, null);
+        return !(
+            kpxcFields.isElementInvisible(fStyle) &&
+            kpxcFields.isElementOpaqueEnough(fStyle) &&
+            kpxcFields.isElementHasValidSize(fStyle) &&
+            !kpxcFields.isElementClipped(fStyle) 
+        )})) {
         return false;
     }
 
     return true;
 };
 
+kpxcFields.isElementInvisible = function(fStyle) {
+    return fStyle.visibility !== 'hidden' && fStyle.visibility !== 'collapse';
+}
+
+kpxcFields.isElementOpaqueEnough = function(fStyle) {
+
+    return Number(fStyle.opacity) >= MIN_OPACITY && Number(fStyle.opacity) <= MAX_OPACITY;
+}
+
+kpxcFields.isElementHasValidSize = function(fStyle) {
+    return parseInt(fStyle.width, 10) > MIN_INPUT_FIELD_WIDTH_PX && parseInt(fStyle.height, 10) > MIN_INPUT_FIELD_WIDTH_PX;
+}
+
+kpxcFields.isElementClipped = function(eStyle) {
+    const clip = eStyle.clip;
+    const clipPath = eStyle.clipPath;
+    const position = eStyle.position;
+    const isClipped = kpxcFields.isClipped(clip, position);
+    const isClippedPath = kpxcFields.isClippedPath(clipPath);
+
+    if (isClipped || isClippedPath) {
+        return true;
+    }
+
+    return false;
+}
+
+kpxcFields.isClipped = function(clip, position) {
+    if (clip !== "auto" && clip.trim() !== "" && (position === "absolute" || position === "fixed")) {
```

Comment:
The regex below this is not very readable. Is there any way to make it more simple? Also, you should use single quotes in the code instead of double quotes.